Add a docs page for forms. (#361)
18F/federalist.18f.gov: pages/documentation/forms.md

---
title: Forms
permalink: /documentation/forms/
layout: page
sidenav: documentation
---

> This page will help you navigate form rules and choose what technology is appropriate.

Reviewer: Suggestion: what -> which
"... which technology is appropriate."

## Form rules
A number of rules may apply to your federal agency's digital form. 

### The Paperwork Reduction Act (PRA)

> The Paperwork Reduction Act determines how federal agencies collect information from the public. Adhering to this law doesn't have to be complicated: a fantastic guide has been written at [https://pra.digital.gov/](https://pra.digital.gov/) to help you understand if any how PRA is applicable to your form. Here's a quick preview of the basic steps you should take:

Reviewer 2: ```suggestion
The Paperwork Reduction Act determines how federal agencies collect information from the public. Adhering to this law doesn't have to be complicated: a fantastic guide has been written at [https://pra.digital.gov/](https://pra.digital.gov/) to help you understand if and how PRA is applicable to your form. Here's a quick preview of the basic steps you should take:
```

1. To figure out if PRA applies to your form, first see if an exemption or generic clearance for common forms applies here: [https://pra.digital.gov/do-i-need-clearance/](https://pra.digital.gov/do-i-need-clearance/)

> 2. If you do need to get PRA approval, then start early! [https://pra.digital.gov/clearance-process/](https://pra.digital.gov/clearance-process/) In some cases it can take 6 to 9 months to get a form approved for your website. To avoid delaying your work, PRA experts recommend you first try to publish a more-basic version of your form that falls under a PRA exemption, while you wait for your full version to be reviewed.

Reviewer 2: ```suggestion
2. If you do need to get PRA approval, then start early! [https://pra.digital.gov/clearance-process/](https://pra.digital.gov/clearance-process/) In some cases it can take 6 to 9 months to get a form approved for your website. To avoid delaying your work, PRA experts recommend you first try to publish a more basic version of your form that falls under a PRA exemption, while you wait for your full version to be reviewed.
```

- [The USWDS form elements](https://designsystem.digital.gov/components/form-controls/)
- A few pre built [form templates](https://designsystem.digital.gov/components/form-templates/).

The USWDS has [508](https://www.section508.gov/manage/laws-and-policies) compliance built in. If you end up creating a form without using the USWDS, you'll need to do the work to make your form accessible. Use [https://accessibility.18f.gov/](https://accessibility.18f.gov/) as a guide.

#### Privacy
Any forms collecting Personally Identifiable Information (PII) will have to comply with their agency's Privacy process. Here is GSA's for reference: [https://before-you-ship.18f.gov/privacy/](https://before-you-ship.18f.gov/privacy/).

#### SORN

> The Federal Records Act also requires that federal agencies publish a list of which systems they keep PII on. These are call System of Record Notices. Each agencies publishes a list, [here](https://www.gsa.gov/reference/gsa-privacy-program/systems-of-records-privacy-act/system-of-records-notices-sorns-privacy-act) is GSA's for reference.

Reviewer 2: ```suggestion
The Federal Records Act also requires that federal agencies publish a list of which systems they keep PII on. These are call System of Record Notices. Each agency publishes a list, [here](https://www.gsa.gov/reference/gsa-privacy-program/systems-of-records-privacy-act/system-of-records-notices-sorns-privacy-act) is GSA's for reference.
```

## Choosing a form service

> Federalist pages are static websites; there is no backend for you to log into to get your form data. You'll need to use a separate form service to use a form on your webpage. Choosing a service depends on what the purpose of your form is. Here are some recommendations.

Reviewer 2: ```suggestion
Federalist only hosts "static" websites; so there is no place to collect form submissions, you'll need to use a separate service to gather the responses from your form. Choosing a service depends on what the purpose of your form is. Here are some recommendations.
```

Author: @davemcorwin Thanks for the edits and suggestions! I included all your commits, though with all the feedback we're getting we will probably rewrite most of this.

#### Google Forms
Google Forms are easy to implement, secure, and customizable enough to handle whatever your form needs are. Here is advice on getting access to Google Forms and for adding them to your website with lots of examples.

<div class="usa-alert usa-alert-warning">
  <div class="usa-alert-body">
    <h3 class="usa-alert-heading">Some agencies can't use Google.</h3>
    <p class="usa-alert-text">Some agencies have the Google Forms domains blocked on their office networks. Unfortunately these agencies will need to find another service to use.</p>
  </div>
</div>

Other agencies, while they may not have an official Google GSuite account purchased for use, can still usually get Google Forms another way.

##### Options for getting access to Google Forms
* Create a Google Account with [your government email](https://accounts.google.com/SignUpWithoutGmail).
* Federalist staff can help set you up a with GSA Affiliated Customer Account (GACA).
* Get a special waiver from your own agency.

##### Advice for using a Google Form

> * Google Forms have a setting that lets you restrict submissions to just users within your agency. Turn that on or off as appropriate for your form.

Reviewer: Suggestion: have -> has
"Google Forms has a setting ..."

For singular reference: https://www.google.com/forms/about/

> * Text fields have a validation option explained [here](https://support.google.com/docs/answer/3378864?hl=en). This can be used to limit submissions to users with a `.gov` or `.mil` email address, as in this [example](https://docs.google.com/forms/d/e/1FAIpQLSePimoF0RkiCP62BSIL_yj0yMXEUePNJ9AabPJqq1Xzbp_GVg/viewform).

Reviewer: The validation options for text fields are explained [here] (...)

---
title: Forms
permalink: /documentation/forms/
layout: page
sidenav: documentation
---
This page will help you navigate form rules and choose what technology is appropriate.

## Form rules
A number of rules may apply to your federal agency's digital form. 

### The Paperwork Reduction Act (PRA)

> The Paperwork Reduction Act (PRA) determines how federal agencies collect information from the public. Adhering to this law doesn't have to be complicated: a fantastic guide has been written at [https://pra.digital.gov/](https://pra.digital.gov/) to help you understand if any how PRA is applicable to your form. Here's a quick preview of the basic steps you should take:

Reviewer 3: the section tone read to me as alerting people to expect PRA to be painful, which I think would problematize something that isn't necessarily or doesn't have to be. So I tried to add more neutral language.

1. To figure out if PRA applies to your form, first see if an exemption or generic clearance for common forms applies here:[https://pra.digital.gov/do-i-need-clearance/](https://pra.digital.gov/do-i-need-clearance/)
2. If you do need to get PRA approval, then start early! [https://pra.digital.gov/clearance-process/](https://pra.digital.gov/clearance-process/)In some cases it can take 6 to 9 months to get a form approved for your website. To avoid delaying your work, PRA experts recommend you first try to publish a more-basic version of your form that falls under a PRA exemption, while you wait for your full version to be reviewed. 
3. After you've started the PRA process, then build the form.
4. Don't go it alone; get in touch with your agency's PRA contact. [https://pra.digital.gov/contact/](https://pra.digital.gov/contact/)

### Privacy Act (PA)

> Add overview and link

Reviewer 3: It would be good to include our learnings on PA.

### Additional rules that may apply
- The 21st Century IDEA Act applies to all executive agencies' public-facing websites. Use USDWS code to build your form found [here] (https://designsystem.digital.gov/components/form-templates/) and [here] (https://designsystem.digital.gov/components/form-controls/), which have important usability guidance baked right in.  

> - While the PRA and PA primarily focus on how data is collected, it is the [Federal Records Act] (https://www.archives.gov/records-mgmt/faqs/federal.html) that focus on how that data is stored or destroyed. Contact [X] to find out more.

Reviewer 3: Tracking down any guidance or contact to list would be good here.

> - If you will be collecting sensitive health data, fill in with overview and link. Contact [X] to find out more.

Reviewer 3: We got info on this in an email thread that I think may have some good stuff to share.

## Choosing a form service
Federalist pages are static websites; there is no backend for you to log into to get your form data. You'll need to use a separate form service to use a form on your webpage. Choosing a service depends on what the purpose of your form is. Here are some recommendations.

> - Newsletter Sign Ups - [GovDelivery](https://granicus.com/solution/govdelivery/) is the most common tool used for newsletters sign ups. Your agency probably already has an account.

Reviewer 3: is it the most common?

Reviewer 3: can we also add in Hubspot?

Author: Yes, GovDelivery is near ubiquitous. I'll be sure to mention Hubspot.

- Customer Feedback - [Touchpoints](https://touchpoints.digital.gov) offers fast track PRA approval for a standardized customer feedback survey. This service costs X.

> - Everything else - [Google Forms](https://gsuite.google.com/products/forms/). Google Forms can handle most all of your form needs for free. Below is more advice for Google Forms.

Reviewer 3: I'm not sure we know enough to say Google is good for everything else, could we back down one step from an absolute statement here and maybe list a couple other common use cases instead that we learned of being used by Google?

#### Google Forms
Google Forms are easy to implement, secure, and customizable enough to handle whatever your form needs are. Here is advice on getting access to Google Forms and for adding them to your website with lots of examples.

<div class="usa-alert usa-alert-warning">
  <div class="usa-alert-body">
    <h3 class="usa-alert-heading">Some agencies can't use Google.</h3>
    <p class="usa-alert-text">Some agencies have the Google Forms domains blocked on their office networks. Unfortunately these agencies will need to find another service to use.</p>
  </div>
</div>

> Most other agencies, while they may not have an official Google GSuite account purchased for use, can still usually get Google Forms another way.

Reviewer 3: again, the word Most just doesn't seem to be supported by what we know. Can we back down from that strong of a statement? Maybe we could address here how some agencies are fully blocked and offer them a link to quickly test for themselves. I believe there was one in the email thread I DMd you on GACA.

##### Options for getting access to Google Forms
* Create a Google Account with your government email [using this form](https://accounts.google.com/SignUpWithoutGmail).
* Federalist staff can help set you up a with GSA Affiliated Customer Account (GACA).

> * Get a special waiver from their own agency.

Reviewer 3: I bet folks would be interested to know more here, maybe a little more context.

Author: I'll get some confirmation from Federalist if they've even done it before, or are willing.

##### Advice for using a Google Form
Google Forms have a setting that lets you restrict submissions to just users within your agency. Turn that on or off as appropriate for your form.

The easiest way to include a Google Form on your page is to just link to it. Here is an [example](https://coe.gsa.gov/connect/contact-us.html#joinus) that links to a few different Google Forms.

To maintain a consistent user experience and design for your forms, we recommend a more advanced technique though. You can write your own HTML form that posts the data to Google Forms. Here are some Federalist examples: [Example](https://tech.gsa.gov/work-with-us/#contact), [Example](https://www.afwerx.af.mil/join.html)

Here is [a well written guide](https://blog.webjeda.com/google-form-customize/) for how to write your own HTML form that matches your created Google Form. You'll need to set your form's action url to post to the Google Form url, then set the correct name element for each of your form's questions. 

> #### USDWS advice

Reviewer 3: I think this would go better in rules section, and just note it as actually a suggestion.

title: Forms on Federalist sites
permalink: /documentation/forms/
layout: page
sidenav: documentation
---
Do you need to ask questions of the people visiting your Federalist website? Do you need to ask for their contact info or let them give you feedback? You need a form. This page provides an overview of existing digital form tools and requirements so you can determine which best apply to your agency's situation and get started.

This pages covers:
* [form services](#choosing-a-form-service)
* [Form requirements](#form-requirements)

## Form services
Federalist pages are static websites. This means there is no backend for you to log into to get your agency’s form data. You’ll need to use a separate form service to use a digital form on your agency’s site. Below is a list of service providers identified through research funded by [GSA's 10x program](https://18f.gsa.gov/tags/10x/). This list is continuously updated, please contact [Federalist product owners](x) if you see other service providers that should be added.

#### [Google Forms](https://www.google.com/forms/about/)
- Use cases: Newsletter sign ups, contact us, feedback, bug reporting, support requests, other custom forms

> - Access: Federalist staff can help you determine if you are able to [create a google account with your government email](https://accounts.google.com/SignUpWithoutGmail) or create a GSA Affiliated Customer Account (GACA). Some agencies are fully blocked from using Google Forms; check with your agency if a special waiver for Google GSuite access may be issued.

Reviewer: Federalist team does not support the determination of google acounts? Perhaps the user should contact their IT department???

Author: We changed that language, telling them to talk to their CIO. 😬

- Quick tips:
  * [Limit submissions to users with a `.gov` or `.mil` email address](https://docs.google.com/forms/d/e/1FAIpQLSePimoF0RkiCP62BSIL_yj0yMXEUePNJ9AabPJqq1Xzbp_GVg/viewform) or certain agency affiliation, see how in these [text validation instructions](https://support.google.com/docs/answer/3378864?hl=en).
  * Link one, or [multiple](https://coe.gsa.gov/connect/contact-us.html#joinus), forms on your page.
  * If you need to include government employees whose agency has blocked Google services, provide an alternative submission method, such as an email address.

> - To get started: Talk to [Federalist product owners](x) to find out if your agency allows access to Google services.

Reviewer: Federalist team does not support the determination of google accounts? Perhaps the user should contact their IT department???

To get started: X. 

#### Privacy Act
What it is: The [Privacy Act](https://www.archives.gov/about/laws/privacy-act-1974.html) requires the government to keep track of any forms collecting Personally Identifiable Information (PII) and to help keep that information secure. If your form is collecting PII from the public, you'll most likely need to:
  - Include Privacy Act Notice language with the relevant form. Here is [GSA's Privacy and Security Notice](https://www.gsa.gov/website-information/privacy-and-security-notice), which is linked to from each form on GSA's website.
  - Ensure the form tool you are using has an official System of Record Notice (SORN). Here is the list of [GSA's SORNs for reference](https://www.gsa.gov/reference/gsa-privacy-program/systems-of-records-privacy-act/system-of-records-notices-sorns-privacy-act).
  - Complete a Privacy Impact Assessment (PIA) to ensure that privacy issues and protections are addressed. Here is a [PIA written for GovDelivery by the Department of the Interior](https://www.doi.gov/sites/doi.gov/files/uploads/govdelivery_pia_final_05.31.2017_1.pdf).

How to get started: Get in contact with your agency's Privacy Office.

> ## Want to learn more about Federalist or forms?

Reviewer: we are not able to support talking to folks about forms but they can come to this docs for guidance on forms.  and we can support maintaining this documentation to provide such guidance. if they have interest in federalist, our sales contact is federalist-iquiries@gsa.gov. information to learn more about federalist is throughout the federalist site ... my thoughts are to remove this section

Author: Hear you, we pulled it.
